src/connector/src/sink/file_sink/mod.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod fs;
+pub mod gcs;
+pub mod opendal_sink;
+pub mod s3;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use arrow_schema::{DataType as ArrowDataType, Field as ArrowField, SchemaRef};
+use async_trait::async_trait;
+use opendal::{Operator, Writer as OpendalWriter};
+use parquet::arrow::AsyncArrowWriter;
+use parquet::file::properties::WriterProperties;
+use risingwave_common::array::{to_record_batch_with_schema, Op, StreamChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::catalog::Schema;
+
+use crate::sink::catalog::SinkEncode;
+use crate::sink::{Result, SinkError, SinkWriter};
+
+// Todo(wcy-fdu): upgrade to opendal 0.47.
+const SINK_WRITE_BUFFER_SIZE: usize = 16 * 1024 * 1024;
+
+pub struct OpenDalSinkWriter {
```

Comment:
nits: we put `OpenDalSinkWriter` in `file_sink/mod.rs` but put `FileSink` in `file_sink/opendal_sink.rs`. it looks a little bit strange.

Reply:
Let's put all those implementation in `opendal_sink.rs`.